### PR TITLE
Improve DDB transaction validation

### PR DESCRIPTION
**Pull request by rafcio19**
Solution for https://github.com/spulec/moto/issues/5496

### CODE REVIEW

File: tests/test_dynamodb/exceptions/test_dynamodb_exceptions.py [hunk 1]
```diff
@@ -761,3 +761,42 @@ def test_query_begins_with_without_brackets():
...
+                }
+            ]
+        )
+        # Verify
+        err = exc.value.response["Error"]
+        assert err["Code"] == "ValidationException"
+        assert (
+            err["Message"]
+            == "TransactItems can only contain one of Check, Put, Update or Delete"
+        )
```
Comment (bobbyi) [inline tests/test_dynamodb/exceptions/test_dynamodb_exceptions.py:802]:
```suggestion
    # Verify
    err = exc.value.response["Error"]
    assert err["Code"] == "ValidationException"
    assert (
        err["Message"]
        == "TransactItems can only contain one of Check, Put, Update or Delete"
)
```

As it stands, if `transact_write_items` throws an exception, the verification code is never run.

Comment (rafcio19) [reply tests/test_dynamodb/exceptions/test_dynamodb_exceptions.py:802]: Good catch

File: moto/dynamodb/exceptions.py [hunk 2]
```diff
@@ -323,3 +323,12 @@ class InvalidConversion(JsonRESTError):
     def __init__(self):
         er = "SerializationException"
         super().__init__(er, "NUMBER_VALUE cannot be converted to String")
+
+
+class TransactWriteSingleOpException(MockValidationException):
+    there_can_be_only_one = (
```
Comment (bblommers) [inline moto/dynamodb/exceptions.py:329]: Nice reference :)

File: moto/dynamodb/models/__init__.py [hunk 3]
```diff
@@ -1655,7 +1656,17 @@ def check_unicity(table_name, key):
             target_items.add(item)
 
         errors = []  # [(Code, Message, Item), ..]
+        ddb_verbs = ["ConditionCheck", "Put", "Delete", "Update"]
         for item in transact_items:
+            # check transact writes are not performing multiple operations
+            # in the same item
+            for ddb_verb in ddb_verbs:
```
Comment (bblommers) [inline moto/dynamodb/models/__init__.py:1663]: This logic doesn't seem quite right. It is possible to have different verbs appear, they just should not act on the same item.
An alternative solution that may work:
List the values in `transact_items`, so the items that are being PUT/DELETED/etc., and then verify there are no duplicates there.
The easiest solution to check for duplicates is `list(values) != set(values)` - the `set`-operation will eliminate any duplicates, so if the list contains more items - we know it's a dup.

**Review by bblommers — CHANGES_REQUESTED** [review]

File: moto/dynamodb/models/__init__.py (hunk 3 above)
Comment (bblommers) [reply moto/dynamodb/models/__init__.py:1663]: Ah, interesting. I completely misunderstood the issue, then. Thanks for the explanation.

Can we still use the `list(keys) == set(keys)`-approach, to verify there are no multiple verbs? That is more readable then iterating over the list twice, IMO

Comment (bblommers) [reply moto/dynamodb/models/__init__.py:1663]: Looks like our replies crossed - using `len(set(..))` would also work, yes :)

**Review by bblommers — APPROVED** [review]
Awesome - thanks for the fix @rafcio19!

**Pull request merged**